linera-rpc/src/simple/server.rs

```diff
@@ -135,7 +136,11 @@ where
         }
     }
 
-    pub fn spawn(self, shutdown_signal: CancellationToken) -> ServerHandle {
+    pub fn spawn(
+        self,
+        shutdown_signal: CancellationToken,
+        tasks: &mut JoinSet<()>,
```

Comment:
nit: `tasks.spawn_task` below is not amazing so perhaps revert to the standard naming practice: `join_set: JoinSet` ?

Reply:
I guess the point of duplicating the type name as the variable name is that any interesting information should be contained within the type… but unfortunately that's not the case here.  Rather, I think it would be better to take the opportunity to specify exactly _which_ tasks are contained in this `JoinSet`, so that the reader knows what happens when the set is joined upon.

Reply:
Done.

Reply:
(note: I missed @Twey's comment. I agree it could be better, but I don't want to bikeshed it now).